File: src/main/java/com/bazbatlabs/bricksmash/models/WorldEvents.java
package com.bazbatlabs.bricksmash.models;

import java.util.Queue;
import java.util.LinkedList;

public final class WorldEvents {

    private final Queue<Event> queue;

    public WorldEvents() {
        this.queue = new LinkedList<Event>();
    }

    public void enqueue(Event event) {
        queue.add(event);
    }

    public Event dequeue() {
        return queue.remove();
    }

    public boolean hasNext() {
        return !queue.isEmpty();
    }

    public static enum Event {
        WALL_HIT,
        PADDLE_HIT,
        BRICK_HIT
    }
}

File: src/main/scala/models/WorldEvents.scala
package com.bazbatlabs.bricksmash.models

import java.util.Queue
import java.util.LinkedList

class WorldEvents {

  val queue = new LinkedList[Event.Value]()

  def enqueue(event: Event.Value) {
    queue.add(event)
  }

  def dequeue(): Event.Value = queue.remove()

  def hasNext = !queue.isEmpty
}

object Event extends Enumeration {
  type Event = Value
  val WallHit, PaddleHit, BrickHit = Value
}
